src/osemgrep/reporting/Cli_json_output.ml

```diff
@@ -345,19 +345,41 @@ let cli_match_of_core_match ~dryrun (hrules : Rule.hrules) (m : OutJ.core_match)
       in
       let fixed_lines =
         if dryrun then
-          Option.map
-            (fun fix ->
-              match (lines, List.rev lines) with
-              | line :: _, last_line :: _ ->
-                  let first_line_part = Str.first_chars line (start.col - 1)
-                  and last_line_part =
-                    Str.string_after last_line (end_.col - 1)
-                  in
-                  String.split_on_char '\n'
-                    (first_line_part ^ fix ^ last_line_part)
-              | [], _
-              | _, [] ->
-                  [])
+          Option.fold ~none:None
```

Comment:
probably better to move this code in a separate helper function.
Also I'm not a fan of those Option.fold, Option.value ~default ...
in general I find the code that does explciit 'match x with Some y -> ... | None -> ...' clearer that those Option.fold.

Reply:
Option.iter is fine (better than Option.fold ~note:()), 
let* is fine.
But I often find Option.fold and Option.value making the code more complicated.

Reply:
done in [a943e44](https://github.com/semgrep/semgrep/pull/9648/commits/a943e44754bd4c3cf6bcd5ef96c612d877603677)